packages/pwa-kit-extension-sdk/src/react/classes/ApplicationExtension.tsx

```diff
@@ -80,7 +83,7 @@ export class ApplicationExtension<
      * @protected
      * @returns a promise resolving to new routes to be added
      */
-    public getRoutesAsync?(params: GetRoutesParams): Promise<RouteProps[]>
+    public getRoutesAsync?(params: GetRoutesParams): Promise<AsyncRouteProps[]>
```

Comment:
`getRoutesAsync` now returns a AsyncRouteProps type that contains `componentProps`